### ISSUE TITLE
PuppetManifest: extras=+q does not generate scoped tags after encountering ensure_packages

### ISSUE THREAD

**Issue by ahakanbaba** [universal-ctags/ctags#1991]
The name of the parser: puppetManifest

The command line you used to run ctags:

```
ctags --options=NONE  --extras=+q --_scopesep-PuppetManifest=*/*::: --_scopesep-PuppetManifest=/*:::  -o - dummy.pp

```

The content of input file:

```
class myClass {
  ensure_packages(['ksh','openssl'], {'ensure' => 'present'})
  $my_variable= 1209600
}
```

The tags output you are not satisfied with:

```
::myClass	dummy.pp	/^class myClass {$/;"	c
::my_variable	dummy.pp	/^  $my_variable= 1209600$/;"	v
myClass	dummy.pp	/^class myClass {$/;"	c
my_variable	dummy.pp	/^  $my_variable= 1209600$/;"	v
```

The tags output you expect:

```
::myClass::my_variable	dummy.pp	/^  $my_variable= 1209600$/;"	v	class:::myClass
```

[ensure_packages](https://forge.puppet.com/puppetlabs/stdlib#ensure_packages) is a valid puppet function. 
After using the `ensure_packages` function, ctags does not generate the scoped tags correctly. 
I would have expected the `::myClass::my_variable` to be emitted regardless of the `ensure_pacakges` usage

Without `ensure_packages` the input looks like this 

```
class myClass {
  # ensure_packages(['ksh','openssl'], {'ensure' => 'present'})
  $my_variable= 1209600
}
```

The tags output is like this: 
```
$ ctags --options=NONE  --extras=+q --_scopesep-PuppetManifest=*/*::: --_scopesep-PuppetManifest=/*:::  -o - dummy.pp
ctags: Notice: No options will be read from files or environment
::myClass	dummy.pp	/^class myClass {$/;"	c
::myClass::my_variable	dummy.pp	/^  $my_variable= 1209600$/;"	v	class:::myClass
myClass	dummy.pp	/^class myClass {$/;"	c
my_variable	dummy.pp	/^  $my_variable= 1209600$/;"	v	class:::myClass
```

The version of ctags:

```
$ ctags --version
Universal Ctags 0.0.0(3ec6f345), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Jan 27 2019, 22:00:52
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +xpath, +yaml, +case-insensitive-filenames
```

How do you get ctags binary:
Building it locally,  from separator-for-optlib @ masatake

The input file is valid puppet code: 
```
$ puppet apply --noop dummy.pp
Notice: Compiled catalog for mbp-005063.ad.whirl.net in environment production in 0.02 seconds
Notice: Applied catalog in 0.02 seconds
```

**Comment by masatake** [comment]
Very strange.

**Comment by masatake** [comment]
Fix is pushed to #1975.

**Comment by masatake** [comment]
I think this bug is fixed via #1975. If not, please, reopen this.
